src/LeakyBucket/Redis.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Jitendra\PhpValve\LeakyBucket;
+
+class Redis extends Base
+{
+    /**
+     * @var \Predis\Client
+     */
+    protected $redis;
+
+    public function __construct(int $maxBucketSize, int $leakRateValue, int $leakRateDuration, \Predis\Client $redis = null)
+    {
+        parent::__construct($maxBucketSize, $leakRateValue, $leakRateDuration);
+
+        $this->redis = $redis ?: new \Predis\Client;
+    }
+
+    public function attempt(string $resource, int $worth = 1): array
+    {
+        $args = [
+            file_get_contents(__DIR__ . '/redis.lua'),
+            2,
```

Comment:
what is 2 here?

Reply:
Ref https://redis.io/commands/eval, 2 is the number of redis keys which follows in the list of args. In eval you have to specify this.